openmc/surface.py

```diff
@@ -94,6 +94,22 @@ def __repr__(self):
 
         return string
 
+    def __key(self):
+        return (self.type, self.boundary_type) + \
+                tuple(j for i in self.coefficients.items() for j in i)
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, other):
+        if isinstance(other, type(self)):
+            return self.__key() == other.__key()
```

Comment:
Thanks for the link and comments @drewejohnson. Would you rather see the key essentially generated in the __hash__ method and the __key method removed entirely? Alternatively I could change __key to _key so that it doesn't invoke the name mangling algorithm.

Reply:
I think it would be better to have the hash method not rely on `_key`, especially if `__hash__` is the only method that uses `_key`.

Reply:
Sounds good.